src/infra/flow-system-inmem/tests/tests/test_flow_service_inmem.rs

```diff
@@ -414,6 +432,237 @@ async fn test_manual_trigger() {
+                    harness.advance_time(Duration::try_milliseconds(100).unwrap()).await;
+                };
+
+                tokio::join!(task0_handle, task2_handle, trigger0_handle, trigger1_handle, main_handle)
+            } => Ok(())
+    }
+    .unwrap();
+
+    pretty_assertions::assert_eq!(
+        format!("{}", test_flow_listener.as_ref()),
+        indoc::indoc!(
+            r#"
+            #0: +0ms:
+
+            #1: +10ms:
+              "foo" HardCompacting:
+                Flow ID = 0 Waiting Manual Executor(task=0, since=10ms)
+
+            #2: +10ms:
+              "foo" HardCompacting:
+                Flow ID = 0 Running(task=0)
+
+            #3: +20ms:
+              "foo" HardCompacting:
+                Flow ID = 0 Finished Success
+
+            #4: +40ms:
+              "bar" HardCompacting:
+                Flow ID = 1 Waiting Manual Executor(task=1, since=40ms)
+              "foo" HardCompacting:
+                Flow ID = 0 Finished Success
+
+            #5: +60ms:
+              "bar" HardCompacting:
+                Flow ID = 1 Running(task=1)
+              "foo" HardCompacting:
+                Flow ID = 0 Finished Success
+
+            #6: +70ms:
+              "bar" HardCompacting:
+                Flow ID = 1 Finished Success
+              "foo" HardCompacting:
+                Flow ID = 0 Finished Success
+
+            "#
+        )
+    );
+}
+
+/////////////////////////////////////////////////////////////////////////////////////////
+
+#[test_log::test(tokio::test)]
+async fn test_manual_trigger_compacting_with_config() {
+    let max_slice_size = 1_000_000u64;
+    let max_slice_records = 1000u64;
+    let harness = FlowHarness::new();
+
+    let foo_id = harness.create_root_dataset("foo").await;
+
+    harness.eager_dependencies_graph_init().await;
+    harness
+        .set_dataset_flow_compacting_rule(
+            harness.now_datetime(),
+            foo_id.clone(),
+            DatasetFlowType::HardCompacting,
+            CompactingRule::new_checked(max_slice_size, max_slice_records).unwrap(),
+        )
+        .await;
+
+    let foo_flow_key: FlowKey =
+        FlowKeyDataset::new(foo_id.clone(), DatasetFlowType::HardCompacting).into();
+
+    let test_flow_listener = harness.catalog.get_one::<FlowSystemTestListener>().unwrap();
+    test_flow_listener.define_dataset_display_name(foo_id.clone(), "foo".to_string());
+
+    // Remember start time
+    let start_time = harness
+        .now_datetime()
+        .duration_round(Duration::try_milliseconds(SCHEDULING_ALIGNMENT_MS).unwrap())
+        .unwrap();
+
+    // Run scheduler concurrently with manual triggers script
+    tokio::select! {
+        // Run API service
+        res = harness.flow_service.run(start_time) => res.int_err(),
+
+        // Run simulation script and task drivers
+        _ = async {
+                // Task 0: "foo" start running at 10ms, finish at 20ms
```

Comment:
Careful with these comments. They are only confusing, if you don't reread and update them